### PR TITLE
Added a check than no more than a fixed number of csw input can be ac…

### PR DESCRIPTION
**Pull request by alsala**
…cepted in the mempool for a SC

### CODE REVIEW

**Review by i-Alex — COMMENTED** [review]
Fine for me. Will accept after tests updates.

File: qa/rpc-tests/sc_csw_eviction_from_mempool.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+# Copyright (c) 2018 The Zencash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.authproxy import JSONRPCException
+from test_framework.util import assert_equal, assert_true, assert_false, initialize_chain_clean, \
+    stop_nodes, wait_bitcoinds, \
+    start_nodes, sync_blocks, sync_mempools, connect_nodes_bi, disconnect_nodes, mark_logs, \
+    dump_sc_info_record, get_epoch_data, swap_bytes, advance_epoch
+from test_framework.mc_test.mc_test import *
```
Comment (ptagl) [inline qa/rpc-tests/sc_csw_eviction_from_mempool.py:12]: I would avoid importing unused symbols.

```suggestion
from test_framework.mc_test.mc_test import CertTestUtils, CSWTestUtils, generate_random_field_element_hex
```

File: qa/rpc-tests/sc_csw_eviction_from_mempool.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+            "scProof": csw_proofs1[-1]
+        }]
+
+        rawtx = self.nodes[0].createrawtransaction([], sc_csw_tx_outs, sc_csw)
+        funded_tx = self.nodes[0].fundrawtransaction(rawtx)
+        sigRawtx = self.nodes[0].signrawtransaction(funded_tx['hex'], None, None, "NONE")
+        mark_logs("Sending a further csw input to scid1, expecting failure...", self.nodes, DEBUG_MODE)
+        try:
+            finalRawtx2 = self.nodes[0].sendrawtransaction(sigRawtx['hex'])
+            assert(false)
```
Comment (ptagl) [inline qa/rpc-tests/sc_csw_eviction_from_mempool.py:257]: Typo, `false` is not defined.

```suggestion
            assert(False)
```

File: qa/rpc-tests/sc_csw_eviction_from_mempool.py [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+        inv_hash = self.nodes[0].getbestblockhash()
+
+        mark_logs("Re-sending previously rejected tx with one more csw...", self.nodes, DEBUG_MODE)
+        try:
+            finalRawtx2 = self.nodes[0].sendrawtransaction(sigRawtx['hex'])
+            mark_logs("sent csw tx {}".format(finalRawtx), self.nodes, DEBUG_MODE)
+        except JSONRPCException, e:
+            errorString = e.error['message']
+            mark_logs("Send csw failed with reason {}".format(errorString), self.nodes, DEBUG_MODE)
+            assert(false)
```
Comment (ptagl) [inline qa/rpc-tests/sc_csw_eviction_from_mempool.py:276]: Typo, `false` is undefined.

```suggestion
            assert(False)
```

**Review by ptagl — CHANGES_REQUESTED** [review]

**Review by ptagl — APPROVED** [review]

File: qa/rpc-tests/sc_csw_eviction_from_mempool.py (hunk 1 above)
Comment (alsala) [reply qa/rpc-tests/sc_csw_eviction_from_mempool.py]: Done

File: qa/rpc-tests/sc_csw_eviction_from_mempool.py (hunk 2 above)
Comment (alsala) [reply qa/rpc-tests/sc_csw_eviction_from_mempool.py]: Done

File: qa/rpc-tests/sc_csw_eviction_from_mempool.py (hunk 3 above)
Comment (alsala) [reply qa/rpc-tests/sc_csw_eviction_from_mempool.py]: Done

**Review by i-Alex — APPROVED** [review]

**Review by albertog78 — APPROVED** [review]

**Pull request merged**